src/main/java/roomescape/domain/reservation/service/ReservationService.java

```diff
@@ -29,19 +52,90 @@ public List<ReservationResponse> getAll() {
+
+        return ReservationResponse.from(savedReservation);
+    }
+
+    private boolean isAlreadyBooked(final ReservationRequest request) {
+        return reservationRepository.existsByDateAndTimeIdAndThemeId(request.date(), request.timeId(), request.themeId());
+    }
+
+    private Reservation getReservation(final ReservationRequest request) {
+        ReservationTime reservationTime = gerReservationTime(request);
+        Theme theme = getTheme(request);
+
+        return Reservation.withoutId(request.name(), request.date(), reservationTime, theme);
+    }
+
+    private Theme getTheme(final ReservationRequest request) {
+        Long themeId = request.themeId();
+        return themeRepository.findById(themeId)
+                .orElseThrow(() -> new EntityNotFoundException("theme not found id =" + themeId));
+    }
+
+    private ReservationTime gerReservationTime(final ReservationRequest request) {
         Long timeId = request.timeId();
-        ReservationTime reservationTime = reservationTimeRepository.findById(timeId)
+        return reservationTimeRepository.findById(timeId)
                 .orElseThrow(() -> new EntityNotFoundException("reservationsTime not found id =" + timeId));
+    }
 
-        Reservation reservation = Reservation.withoutId(request.name(), request.date(), reservationTime);
-
-        Reservation saved = reservationRepository.save(reservation);
+    private void validateDateTime(final LocalDateTime now, final LocalDate date, final LocalTime time) {
+        LocalDateTime dateTime = LocalDateTime.of(date, time);
 
-        return ReservationResponse.from(saved);
+        if (now.isAfter(dateTime)) {
+            throw new IllegalArgumentException("이미 지난 예약 시간입니다.");
+        }
     }
 
-    public void delete(Long id){
+    public void delete(final Long id) {
         reservationRepository.deleteById(id);
     }
+
+    public List<BookedReservationTimeResponse> getAvailableTimes(final LocalDate date, final Long themeId) {
+        Map<ReservationTime, Boolean> allTimes = processAlreadyBookedTimesMap(date, themeId);
+
+        return allTimes.entrySet()
+                .stream()
+                .map(entry -> bookedReservationTimeResponseOf(entry.getKey(), entry.getValue()))
+                .toList();
+    }
+
+    private Map<ReservationTime, Boolean> processAlreadyBookedTimesMap(final LocalDate date, final Long themeId) {
+        Set<ReservationTime> alreadyBookedTimes = getAlreadyBookedTimes(date, themeId);
+
+        return reservationTimeRepository.findAll()
+                .stream()
+                .collect(Collectors.toMap(Function.identity(), time -> isAlreadyBookedTime(time, alreadyBookedTimes)));
+    }
+
+    private Boolean isAlreadyBookedTime(final ReservationTime time, final Set<ReservationTime> alreadyBookedTimes) {
+        if (alreadyBookedTimes.contains(time)) {
+            return IS_ALREADY_BOOKED;
+        }
+        return IS_UNBOOKED;
+    }
```

Comment:
```suggestion
    private Boolean isAlreadyBookedTime(final ReservationTime time, final Set<ReservationTime> alreadyBookedTimes) {
        return alreadyBookedTimes.contains(time);
    }
```

조건문을 바로 return 하면 상수 분리가 필요 없어보여요~